### PR TITLE
Feat/add test case for get boosts

### LINKED ISSUE #603: test: `/boost/get_boosts`
read description here - https://github.com/starknet-id/starknet.quest/issues/602#issue-2255907921

### PR DESCRIPTION
**Pull request by bhavyagosai**
# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Testing

Resolves: #603 

## Other information

my test covers the lines `53-61` ie `getBoosts()` for the file `apiService.ts`

image

### CODE REVIEW

File: app/quest-boost/page.tsx [hunk 1]
```diff
@@ -22,7 +22,7 @@ export default function Page() {
   const fetchBoosts = async () => {
     try {
       const res = await getBoosts();
-      setBoosts(res);
+      if (res) setBoosts(res);
```
Comment (ayushtom) [inline app/quest-boost/page.tsx]: can set this as 
```
if(!res) {
// show error message on screen saying - No boosts available now
(you can use error message screen component in the codebase to do this)
}
setBoosts(res);
```

Comment (bhavyagosai) [reply app/quest-boost/page.tsx]: @ayushtom This is the current error message (No quests are being boosted at the moment)

image

Should I change it to the one below? But then the ErrorScreen component takes the whole page height and there is content below it...

image

Comment (ayushtom) [reply app/quest-boost/page.tsx]: Hmm , actually what you can do is go with the first option, but also keep the same space in the section as the cards were taking and place the text in the center. Currently it looks a little stuck together vertically. wdyt?

Comment (bhavyagosai) [reply app/quest-boost/page.tsx]: makes sense, I was thinking something of the same sort too

**Comment by bhavyagosai** [comment]
- Have added checks wherever `boost/get_boosts` api was being fetched and displayed
- Found one component `QuestCategories` where the api was being used but not sure if currently the component is being used anywhere as I found it only to be declared but never used. Nevertheless, have updated it as well.

Something I wanted to point out was that, currently there are no loaders displayed while any loading occurs while fetching the data through apis. I feel we ought to add a placeholder/loader wherever data is loading.

**Comment by ayushtom** [comment]
Screenshot 2024-04-27 at 2 16 27 PM

Hey i cant see the boosts when i come to this page. Also your suggestion regarding loaders is right. Can you add a loader on this page when we fetch boosts? I think currently the - "No boosts available" text comes up first when i come on this page

**Comment by bhavyagosai** [comment]
> I think currently the - "No boosts available" text comes up first when i come on this page

Yeah exactly. I'll fix them

**Comment by bhavyagosai** [comment]
@ayushtom idk about not being able to see the boosts, I am able to see them. You can confirm this again now I have updated some things.

- added loader in `\quest-boost` page.
- moved the api call functions inside useEffect() as they were being used only inside that scope.
- improved code logic for BoostCard useEffect() check for any/all quests being completed by user

One more thing:
https://github.com/bhavyagosai/starknet.quest/blob/e54f9a762491e468510feb405fe401be60e5f3c5/app/quest-boost/page.tsx#L64 I felt as both boosts and completedQuests array were fetched and displayed here (or used for any logic), so rendering this after they have been fetched. You can correct me if I am wrong here.

Note: have not seen all part of code but maybe small small code improvements can be made elsewhere regarding adding loaders and all 😅

**Review by ayushtom — APPROVED** [review]
lgtm

**Pull request merged**